genesis/utils/path_planing.py

```diff
@@ -0,0 +1,1127 @@
+from abc import ABC, abstractmethod
+from typing import Literal
+import numpy as np
+import taichi as ti
+import torch
+import torch.nn.functional as F
+import math
+import time
+import genesis as gs
+from genesis.utils.misc import tensor_to_array
+
+try:
+    from ompl import base as ob
+    from ompl import geometric as og
+    from ompl import util as ou
+
+    IS_OMPL_AVAILABLE = True
+
+    __all__ = [
+        "RRTConnect",
+        "RRT",
+        "RRTConnect_OMPL",
+        "RRT_OMPL",
+        "PRM_OMPL",
+        "EST_OMPL",
+        "FMT_OMPL",
+        "BITstar_OMPL",
+        "ABITstar_OMPL",
+    ]
+
+except ImportError:
+    IS_OMPL_AVAILABLE = False
+
+    __all__ = [
+        "RRTConnect",
+        "RRT",
+    ]
+
+
+@ti.data_oriented
+class PathPlanner(ABC):
```

Comment:
I don't see the point of doing this. I would recommend explicitly calling `self._entity` and `self._solver`. This is clearer and more consistent with what we are doing elsewhere. Because I'm not convinced these attributes should be part of the public API. But if you think they should and also @YilingQiao, then you can keep the current design.